### PR TITLE
[bug] don't remove timezone-awareness when using the  method from Dat…

### LINKED ISSUE #30238: Appending to DataFrame removes the timezone-awareness
#### Code Sample, a copy-pastable example if possible

```python
import pytz

import pandas as pd

df = pd.DataFrame(
    {"a": [pd.Timestamp("2019-07-19 07:04:57+0100", tz=pytz.FixedOffset(60))]}
)
df.append(df.iloc[0])
```
#### Problem description

This returns:
```
|    | a                         |
|---:|:--------------------------|
|  0 | 2019-07-19 07:04:57+01:00 |
|  0 | 2019-07-19 06:04:57       |
```
However, I was expecting
```
|    | a                         |
|---:|:--------------------------|
|  0 | 2019-07-19 07:04:57+01:00 |
|  0 | 2019-07-19 07:04:57+01:00 |
```
which indeed is what happens if I instead write
```
df.append(df)
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.7.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-72-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 0.25.3
numpy            : 1.17.4
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 19.3.1
setuptools       : 41.6.0.post20191030
Cython           : 0.29.14
pytest           : 5.3.1
hypothesis       : None
sphinx           : 2.2.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.9.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.1
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
s3fs             : 0.4.0
scipy            : 1.3.3
sqlalchemy       : None
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
…aFrame

- [x] closes #30238
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -6765,7 +6765,7 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
                 other.values.reshape((1, len(other))),
                 index=index,
                 columns=combined_columns,
-            )
+            ).astype(other.dtype)
```
Comment (jreback) [inline pandas/core/frame.py]: this is not great way to fix this
the construction needs to change

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (MarcoGorelli) [reply pandas/core/frame.py]: @jreback thanks for your quick review. I tried passing the `dtype` argument directly, but then ran into #30280.

However, I've been able to get around the problem by rewriting the code a bit (as I understand, chaining methods is now the preferred way of using Pandas). The tests pass.

What I don't (yet) understand is the failing CI build:
```
error: Module has no attribute "FixedOffset"
```
Other parts of the code seem to be using
```
import pytz
tz=pytz.FixedOffset
```
without any issues, but I'll look into it

File: pandas/core/frame.py [hunk 2]
```diff
@@ -6748,25 +6748,19 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
...
                 combined_columns = self.columns.append(idx_diff)
             except TypeError:
                 combined_columns = self.columns.astype(object).append(idx_diff)
-            other = other.reindex(combined_columns, copy=False)
-            other = DataFrame(
-                other.values.reshape((1, len(other))),
-                index=index,
-                columns=combined_columns,
+            other = (
+                other.reindex(combined_columns, copy=False)
```
Comment (jreback) [inline pandas/core/frame.py]: instead of ``_convert`` you can use``.infer_object()`` here

File: pandas/tests/frame/test_combine_concat.py [hunk 3]
```diff
@@ -288,6 +289,20 @@ def test_append_dtypes(self):
         expected = DataFrame({"bar": Series([Timestamp("20130101"), 1])})
         tm.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
```
Comment (jreback) [inline pandas/tests/frame/test_combine_concat.py]: instead of this use the tz_naive_fixture which covers a bunch of cases + None

**Review by jreback — CHANGES_REQUESTED** [review]
append is overly complicated and duplicating many parts of concat. 

I think you might be able to remove almost all of 6751-6771, e.g. once its a Series . All of the column stuff is not necessary (I could be wrong but give this a try)

IOW this does the right thing

```
In [17]: df = pd.DataFrame({'A': pd.date_range('20130101', periods=3, tz='UTC'), 'B': range(3)})                                                                                                                            

In [18]: pd.concat([df, df.iloc[0].to_frame().T.infer_objects()]).dtypes                                                                                                                                                    
Out[18]: 
A    datetime64[ns, UTC]
B                  int64
dtype: object
```

**Comment by MarcoGorelli** [comment]
> All of the column stuff is not necessary (I could be wrong but give this a try)

Tried this, but it seems it's necessary for `pandas/tests/reshape/test_concat.py::TestAppend::test_append_same_columns_type`

**Comment by mroeschke** [comment]
@MarcoGorelli this change solved the original issue. Maybe give this a shot:

```
(pandas-dev) matthewroeschke:pandas-mroeschke matthewroeschke$ git diff
diff --git a/pandas/core/frame.py b/pandas/core/frame.py
index 766437dba..62391b7cc 100644
--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -6771,7 +6771,7 @@ class DataFrame(NDFrame):
                 combined_columns = self.columns.astype(object).append(idx_diff)
             other = other.reindex(combined_columns, copy=False)
             other = DataFrame(
-                other.values.reshape((1, len(other))),
+                other._values,
                 index=index,
                 columns=combined_columns,
             )
```

**Comment by MarcoGorelli** [comment]
@mroeschke I would still need to reshape though, wouldn't I? Else I don't see how
```
pandas/tests/frame/test_combine_concat.py::TestDataFrameConcatCommon::test_append_series_dict
```
would pass.

But if I do reshape, then `pandas/tests/frame/test_combine_concat.py::TestDataFrameConcatCommon::test_append_timestamps_aware_or_naive` fails with
```
AttributeError: 'DatetimeArray' object has no attribute 'reshape'
```

**Comment by mroeschke** [comment]
Extension arrays (DatetimeArray) are still essentially 1D only right now, so reshaping is not possible. As you can see in the `test_append_series_dict` it's failing the `verify_integrety=True` test which I think are not really well integrated with extension arrays.

File: pandas/tests/frame/test_combine_concat.py [hunk 4]
```diff
@@ -288,6 +288,17 @@ def test_append_dtypes(self):
...
+        "timestamp", ["2019-07-19 07:04:57+0100", "2019-07-19 07:04:57"]
+    )
+    def test_append_timestamps_aware_or_naive(self, tz_naive_fixture, timestamp):
+        # GH 30238
+        tz = tz_naive_fixture
+        df = pd.DataFrame([pd.Timestamp(timestamp, tz=tz)])
+        result = df.append(df.iloc[0]).iloc[-1]
+        expected = pd.Series(pd.Timestamp(timestamp, tz=tz), name=0)
+        pd.testing.assert_series_equal(result, expected)
```
Comment (jreback) [inline pandas/tests/frame/test_combine_concat.py]: don't directly use ``pd.testing``

``tm.assert_series_equal``

File: pandas/core/frame.py [hunk 5]
```diff
@@ -6757,25 +6757,18 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=False):
...
+                .to_frame()
+                .T.infer_objects()
+                .rename_axis(index.names)
```
Comment (jreback) [inline pandas/core/frame.py]: you can add copy=False here

**Review by jreback — CHANGES_REQUESTED** [review]
small changes. I would *also* like to test this with categorical, nullable integer and other extension types (likely this fixes).

so if you can file an issue for this (and followup PR would be great). 

can also do here if you'd like alternately

**Comment by MarcoGorelli** [comment]
@mroeschke Thanks for your explanation. What would you like me to do then, use `._value` and change `test_append_series_dict`, or keep as is?

**Comment by MarcoGorelli** [comment]
@jreback thanks for your review, I've made those changes.

I've opened up another issue for the other dtypes, will work on that next week if no one else has taken it up by then

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**